frontend/src/components/dashboard/campaigns/Campaigns.tsx

```diff
@@ -63,18 +66,50 @@ const Campaigns = () => {
     setLoading(false)
   }
 
+  // Returns true if lastSeenVersion < currentVersion, else false
+  function compareSemver(currentVersion: string, lastSeenVersion: string) {
+    if (!currentVersion) {
+      return true
+    }
+    const currentSplit = currentVersion
+      .split('.')
+      .map((num) => parseInt(num, 10))
+    const lastSplit = lastSeenVersion.split('.').map((num) => parseInt(num, 10))
+    for (let i = 0; i < 3; i++) {
+      if (currentSplit[i] < lastSplit[i]) {
+        return true
+      }
+      if (currentSplit[i] > lastSplit[i]) {
+        return false
+      }
+    }
+    return false
+  }
+
+  function displayNewAnnouncement(announcementVersion: string) {
+    if (
+      ANNOUNCEMENT.isActive &&
+      compareSemver(announcementVersion, i18n._(ANNOUNCEMENT.version))
+    ) {
+      modalContext.setModalContent(<AnnouncementModal />)
+    }
+  }
+
   useEffect(() => {
     fetchCampaigns(selectedPage)
   }, [selectedPage])
 
   useEffect(() => {
-    async function getNumDemos() {
-      const { demo } = await getUserSettings()
+    // TODO: refactor out num demos processing
+    async function getNumDemosAndAnnouncementVersion() {
+      const { demo, announcementVersion } = await getUserSettings()
       setIsDemoDisplayed(demo?.isDisplayed)
       setNumDemosSms(demo?.numDemosSms)
       setNumDemosTelegram(demo?.numDemosTelegram)
+      displayNewAnnouncement(announcementVersion)
     }
-    getNumDemos()
+    getNumDemosAndAnnouncementVersion()
+    // eslint-disable-next-line
```

Comment:
Is there a dependency we are trying to avoid? Not sure if the infinite re-render issue was resolved from when we last discussed it

Reply:
Discussed this with @lamkeewei -- we decided to implement a check to see if `modalContent` is currently `null` or not. If it is not null, then we'll trigger `setModalContent`. This helps with breaking the loop since the context isn't being constantly updated.